autopilot/percentile_test.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2014-2020 Montana Flynn (https://montanaflynn.com)
+package autopilot
+
+import (
+	"testing"
+)
+
+// TestPercentile is copied from Montana Flynn's stats library since the
+// implementation of percentile is more or less copied. Copying this test
+// asserts both implementations behave the same way.
+func TestPercentile(t *testing.T) {
```

Comment:
feels unnecessary to me; I don't think we have any intention of changing the implementation of `percentile`

Reply:
Usually when you copy an implementation you add a unit test to assert they behave exactly the same way right? It's not an exact copy because I wanted to avoid copying over all of the custom types that Flynn added to his library. It needlessly makes the percentile function a 200 lines file as opposed to the ~30 lines it is now. I can remove it though if you feel it's totally useless.

We weren't going to change the implementation of `randSelectByWeight` either and thanks to Chris that urged to write a unit test for it found it had a bug 🤷‍♂️

Reply:
well, this test has now passed, and it's completely deterministic, which means it's going to run and pass ten thousand more times, providing zero information. This is exactly the sort of test that I want to cut back on. In the unlikely event that we later modify `percentile`, we can test it again at that time.